controllers/yawol-controller/loadbalancermachine/loadbalancermachine_controller.go

```diff
@@ -726,55 +804,76 @@ func (r *LoadBalancerMachineReconciler) deleteServer(
+	var requeue bool
 	var portClient os.PortClient
 	portClient, err = osClient.PortClient(ctx)
 	if err != nil {
-		return err
+		return false, err
 	}
 
-	if lbm.Status.PortID != nil {
-		if err = openstackhelper.DeletePort(ctx, portClient, *lbm.Status.PortID); err != nil {
+	if lbm.Status.DefaultPortID != nil {
+		if err = openstackhelper.DeletePort(ctx, portClient, *lbm.Status.DefaultPortID); err != nil {
 			switch err.(type) {
 			case gophercloud.ErrDefault404:
-				r.Log.Info("error deleting port, already deleted", "lbm", lbm.Name)
+				_ = helper.RemoveFromLBMStatus(ctx, r.Status(), lbm, "defaultPortIP")
+				if err := helper.RemoveFromLBMStatus(ctx, r.Client.Status(), lbm, "defaultPortID"); err != nil {
+					return false, err
+				}
 			default:
 				r.Log.Info("an unexpected error occurred deleting port", "lbm", lbm.Name)
-				return kubernetes.SendErrorAsEvent(r.Recorder, err, lbm)
+				return false, kubernetes.SendErrorAsEvent(r.Recorder, err, lbm)
 			}
 		}
+		requeue = true
 	}
 
-	// delete orphan ports
-	portName := req.NamespacedName.String()
-	var portList []ports.Port
-	portList, err = portClient.List(ctx, ports.ListOpts{Name: portName})
-	if err != nil {
-		return err
-	}
-
-	for i := range portList {
-		port := &portList[i]
-		// double check
-		if port.ID == "" || port.Name != portName {
-			continue
-		}
-
-		if err = openstackhelper.DeletePort(ctx, portClient, port.ID); err != nil {
+	// TODO cleanup after removing deprecated fields
+	if lbm.Status.PortID != nil { //nolint: staticcheck // needed to be backwards compatible
+		//nolint: staticcheck // needed to be backwards compatible
+		if err = openstackhelper.DeletePort(ctx, portClient, *lbm.Status.PortID); err != nil {
 			switch err.(type) {
 			case gophercloud.ErrDefault404:
-				r.Log.Info("error deleting port, already deleted", "lbm", lbm.Name)
+				if err := helper.RemoveFromLBMStatus(ctx, r.Client.Status(), lbm, "portID"); err != nil {
+					return false, err
+				}
 			default:
 				r.Log.Info("an unexpected error occurred deleting port", "lbm", lbm.Name)
-				return kubernetes.SendErrorAsEvent(r.Recorder, err, lbm)
+				return false, kubernetes.SendErrorAsEvent(r.Recorder, err, lbm)
 			}
 		}
+		requeue = true
 	}
 
-	return nil
+	if requeue {
+		return true, nil
+	}
+
+	if lbm.Status.DefaultPortName != nil {
+		port, err := openstackhelper.GetPortByName(ctx, portClient, *lbm.Status.DefaultPortName)
+		if err != nil {
+			return false, err
+		}
+		if port != nil && port.ID != "" {
+			if err = openstackhelper.DeletePort(ctx, portClient, port.ID); err != nil {
+				switch err.(type) {
+				case gophercloud.ErrDefault404:
+				default:
+					r.Log.Info("an unexpected error occurred deleting port", "lbm", lbm.Name)
+					return false, kubernetes.SendErrorAsEvent(r.Recorder, err, lbm)
+				}
+			}
+			requeue = true
+		} else {
+			if err := helper.RemoveFromLBMStatus(ctx, r.Status(), lbm, "defaultPortName"); err != nil {
+				return false, err
+			}
+		}
```

Comment:
```suggestion
		}
		
		if err := helper.RemoveFromLBMStatus(ctx, r.Status(), lbm, "defaultPortName"); err != nil {
			return false, err
		}
```

Reply:
this would remove the portname in the status even when the port is already deleted (errdefault404 case)

Reply:
See comment above. In addition there should be only a port in this state if something went wrong (normale the port should already be deleted via the ID). So requeue and check everything again sound fine for me.